yasa/evaluation.py

```diff
@@ -0,0 +1,649 @@
+        prop_bias_results = []
+        residuals_results = []
+        for ss, ss_df in self.data.groupby(self.statistic):
+            # Regress the difference score on the reference device
+            model = pg.linear_regression(ss_df[self.reference], ss_df["difference"], **kwargs)
+            model.insert(0, self.statistic, ss)
+            # Extract the subject-level residuals
+            resid = pd.DataFrame(
+                {
+                    self.subject: ss_df[self.subject],
+                    self.statistic: ss,
+                    "pbias_residual": model.residuals_
+                }
+            )
+            prop_bias_results.append(model)
+            residuals_results.append(resid)
+        # Add residuals to raw dataframe, used later when testing homoscedasticity
+        residuals = pd.concat(residuals_results)
+        self.data = self.data.merge(residuals, on=[self.subject, self.statistic])
+        # Handle proportional bias results
+        prop_bias = pd.concat(prop_bias_results)
+        # Save all the proportional bias models before removing intercept, for optional user access
+        self.proportional_bias_models_ = prop_bias.reset_index(drop=True)
+        # Remove intercept rows
+        prop_bias = prop_bias.query("names != 'Intercept'").drop(columns="names")
+        # Add True/False passing column for easy access
+        prop_bias["unbiased"] = prop_bias["pval"].ge(kwargs["alpha"])
+        self.proportional_bias = prop_bias.set_index(self.statistic)
+
+    def test_homoscedasticity(self, **kwargs):
+        """Test each statistic for homoscedasticity.
+
+        Parameters
+        ----------
+        **kwargs : key, value pairs
+            Additional keyword arguments are passed to :py:func:`pingouin.homoscedasticity`.
+
+        ..note:: ``self.test_proportional_bias()`` must be run first.
+        """
+        group = self.data.groupby(self.statistic)
+        columns = [self.reference, "difference", "pbias_residual"]
+        homoscedasticity = group.apply(lambda df: pg.homoscedasticity(df[columns], **kwargs))
+        self.homoscedasticity = homoscedasticity.droplevel(-1)
+
+    def summary(self, descriptives=True):
+        """Return a summary dataframe highlighting what statistics pass checks."""
+        assert isinstance(descriptives, bool), "`descriptives` must be True or False"
+        series_list = [
+            self.normality["normal"],
+            self.proportional_bias["unbiased"],
+            self.homoscedasticity["equal_var"].rename("homoscedastic"),
+        ]
+        summary = pd.concat(series_list, axis=1)
+        if descriptives:
+            group = self.data.drop(columns=self.subject).groupby(self.statistic)
+            desc = group.agg(["mean", "std"])
+            desc.columns = desc.columns.map("_".join)
+            summary = summary.join(desc)
+        return summary
+
+    def plot_discrepancies_heatmap(self, sstats_order=None, **kwargs):
```

Comment:
I'll review the remaining (plotting) functions later this week. I really need to try them out to understand them but my regular computer is currently in repair.